ssas/service/admin/api_test.go

```diff
@@ -573,6 +573,187 @@ func (s *APITestSuite) TestJsonError() {
+	_ = json.Unmarshal(rr.Body.Bytes(), &ips)
+	assert.NotEmpty(s.T(), ips)
+	assert.Len(s.T(), ips, 3)
+
+	_ = ssas.CleanDatabase(group)
+}
+
+func (s *APITestSuite) TestGetSystemIPsBadSystem() {
+	//Should not exits
+	badSysId := 42
+
+	systemID := strconv.FormatUint(uint64(badSysId), 10)
+	req := httptest.NewRequest("GET", "/system/"+systemID+"/ip", nil)
+	rctx := chi.NewRouteContext()
+	rctx.URLParams.Add("systemID", systemID)
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+	handler := http.HandlerFunc(getSystemIPs)
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusNotFound, rr.Result().StatusCode)
+}
+
+func (s *APITestSuite) TestRegisterSystemIP() {
+	group := ssas.Group{GroupID: "test-reset-creds-group"}
+	s.db.Create(&group)
+	system := ssas.System{GID: group.ID, ClientID: "test-reset-creds-client"}
+	s.db.Create(&system)
+
+	systemID := strconv.FormatUint(uint64(system.ID), 10)
+	body := "{\"address\":\"192.12.22.81\"}"
+	req := httptest.NewRequest("POST", "/system/"+systemID+"/ip", strings.NewReader(body))
+	rctx := chi.NewRouteContext()
+	rctx.URLParams.Add("systemID", systemID)
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+	handler := http.HandlerFunc(registerIP)
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusCreated, rr.Result().StatusCode)
+	assert.Equal(s.T(), "application/json", rr.Result().Header.Get("Content-Type"))
+
+	//Retrieve to confirm
+	req = httptest.NewRequest("GET", "/system/"+systemID+"/ip", nil)
+	rctx = chi.NewRouteContext()
+	rctx.URLParams.Add("systemID", systemID)
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+	handler = getSystemIPs
+
+	rr = httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusOK, rr.Result().StatusCode)
+	assert.Equal(s.T(), "application/json", rr.Result().Header.Get("Content-Type"))
+	var ips []ssas.IP
+	_ = json.Unmarshal(rr.Body.Bytes(), &ips)
+	assert.Len(s.T(), ips, 1)
+
+	_ = ssas.CleanDatabase(group)
+}
+
+func (s *APITestSuite) TestRegisterInvalidIP() {
+	group := ssas.Group{GroupID: "test-reset-creds-group"}
+	s.db.Create(&group)
+	system := ssas.System{GID: group.ID, ClientID: "test-reset-creds-client"}
+	s.db.Create(&system)
+
+	systemID := strconv.FormatUint(uint64(system.ID), 10)
+	body := "{\"address\":\"600.1\"}"
+	req := httptest.NewRequest("POST", "/system/"+systemID+"/ip", strings.NewReader(body))
+	rctx := chi.NewRouteContext()
+	rctx.URLParams.Add("systemID", systemID)
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+	handler := http.HandlerFunc(registerIP)
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusBadRequest, rr.Result().StatusCode)
+
+	_ = ssas.CleanDatabase(group)
+}
+
+func (s *APITestSuite) TestRegisterMaxSystemIP() {
+	ssas.MaxIPs = 3
```

Comment:
Changing a global for a single unit test can lead to race conditions in parallel tests.  I see that the same value is set again for other tests in this suite; if there aren't side effects in other suites, maybe we could configure it just once in `SetupSuite()`.

Reply:
done.